### PR TITLE
contest: add optional question  "How did you learn about the contest"

### PR DESCRIPTION
**Pull request by fnareoh**
Adresses #42

### CODE REVIEW

**Comment by fnareoh** [comment]
Currently lacking translations

File: prologin/contest/models.py [hunk 1]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
     ugettext_noop("Assigned")
 
 
+class Learn_about_contest(ChoiceEnum):
+    already_knows = (0, _("I already knew it"))
```
Comment (seirl) [inline prologin/contest/models.py]: "I don't remember", maybe? You have to have known it by something at some point.

File: prologin/contest/models.py [hunk 2]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
...
+    web_site = (1, _("By the website"))
```
Comment (seirl) [inline prologin/contest/models.py]: You can't learn about it from the website, you have to know how to go to our website first

File: prologin/contest/models.py [hunk 3]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
...
+    social_medias = (2, _("By social medias"))
```
Comment (seirl) [inline prologin/contest/models.py]: From social media

File: prologin/contest/models.py [hunk 4]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
...
+    poster = (3, _("By the poster"))
```
Comment (seirl) [inline prologin/contest/models.py]: "From the poster in my school/university"?

File: prologin/contest/models.py [hunk 5]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
...
+    word_of_mouth = (4, _("By word of mouth"))
+    svj = (5, _("Magazine science et vie junior"))
```
Comment (seirl) [inline prologin/contest/models.py]: If we go into such specific details, we'll need to update that list depending on the editions. If that's the case, then this needs to be a database table, not something in the code.

File: prologin/contest/models.py [hunk 6]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
     ugettext_noop("Assigned")
 
 
+class Learn_about_contest(ChoiceEnum):
+    already_knows = (0, _("I already knew it"))
+    web_site = (1, _("By the website"))
+    social_medias = (2, _("By social medias"))
+    poster = (3, _("By the poster"))
+    word_of_mouth = (4, _("By word of mouth"))
```
Comment (seirl) [inline prologin/contest/models.py]: "From a friend" maybe?

File: prologin/contest/models.py [hunk 7]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
     ugettext_noop("Assigned")
 
 
+class Learn_about_contest(ChoiceEnum):
```
Comment (seirl) [inline prologin/contest/models.py]: Use CamelCase for classes

File: prologin/contest/models.py [hunk 8]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
...
+    already_knows = (0, _("I already knew it"))
+    web_site = (1, _("By the website"))
+    social_medias = (2, _("By social medias"))
+    poster = (3, _("By the poster"))
+    word_of_mouth = (4, _("By word of mouth"))
+    svj = (5, _("Magazine science et vie junior"))
+    other = (6, _("Other way"))
```
Comment (seirl) [inline prologin/contest/models.py]: "Some other way"

Comment (seirl) [inline prologin/contest/models.py]: Maybe add "Another programming contest"?

File: prologin/contest/models.py [hunk 9]
```diff
@@ -215,7 +229,11 @@ class Contestant(ExportModelOperationsMixin('contestant'), models.Model):
     preferred_language = CodingLanguageField(blank=True, db_index=True,
                                              help_text=_("The programming language you will most likely use during the "
                                                          "regional events."))
-
+    learn_about_contest = EnumField(
+        Learn_about_contest, null=True,
+        blank=True, db_index=True,
+        verbose_name=_("How did you know the contest"),
```
Comment (seirl) [inline prologin/contest/models.py]: s/know/learn about"

Question mark at the end.

**Review by seirl — CHANGES_REQUESTED** [review]

File: prologin/contest/models.py [hunk 10]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
     ugettext_noop("Assigned")
 
 
+class LearnAboutContest(ChoiceEnum):
+    doesnt_know = (0, _("I don't remember"))
+    other_contest = (1, _("Another programming contest"))
+    social_medias = (2, _("From social media"))
+    poster = (3, _("From the poster in my school/university"))
+    word_of_mouth = (4, _("From a friend"))
+    svj = (5, _("From a journal"))
```
Comment (seirl) [inline prologin/contest/models.py]: The field is still called "SVJ". Also "From a journal" isn't very clear, so you should probably expand on that.

File: prologin/contest/models.py [hunk 11]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
     ugettext_noop("Assigned")
 
 
+class LearnAboutContest(ChoiceEnum):
+    doesnt_know = (0, _("I don't remember"))
```
Comment (seirl) [inline prologin/contest/models.py]: I don't know according to what the fields are sorted, but that would be better at the end of the dropdown.

File: prologin/contest/models.py [hunk 12]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
...
+    other_contest = (1, _("Another programming contest"))
+    social_medias = (2, _("From social media"))
```
Comment (seirl) [inline prologin/contest/models.py]: no s at social_media

**Review by seirl — CHANGES_REQUESTED** [review]

File: prologin/contest/models.py [hunk 13]
```diff
@@ -202,6 +202,20 @@ class Assignation(ChoiceEnum):
...
+    poster = (3, _("From the poster in my school/university"))
+    word_of_mouth = (4, _("From a friend"))
```
Comment (seirl) [inline prologin/contest/models.py]: Update the field name to "friend" maybe?

**Review by seirl — CHANGES_REQUESTED** [review]

**Review by seirl — APPROVED** [review]

**Pull request merged**